### PR TITLE
Workload identity submodule

### PR DESCRIPTION
**Pull request by bharathkkb**
Added commits from #307 for Workload Identity
- Added submodule for Workload Identity
- Added example
- Added test

### CODE REVIEW

**Comment by morgante** [comment]
@bharathkkb Please address test errors.

**Comment by bharathkkb** [comment]
@morgante I think the tests are flaky; sandbox-enabled-local for some reason is taking more than 30 mins to bring up the node pools.

**Comment by bharathkkb** [comment]
@morgante test are passing PTAL

File: examples/workload_identity/variables.tf [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  description = "The project ID to host the cluster in"
+}
+
+variable "cluster_name_suffix" {
+  description = "A suffix to append to the default cluster name"
+  default     = ""
+}
+
+variable "region" {
+  description = "The region to host the cluster in"
```
Comment (morgante) [inline examples/workload_identity/variables.tf]: Non-blocker: we should have a default for this.

File: examples/workload_identity/variables.tf [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+}
+
+variable "zones" {
+  type        = list(string)
+  description = "The zone to host the cluster in (required if is a zonal cluster)"
```
Comment (morgante) [inline examples/workload_identity/variables.tf]: Non-blocker: For examples, we can/should be super opinionated (ex. *only* support regional clusters).

File: modules/workload-identity/README.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+## Usage
+
+The `terraform-google-workload-identity` can create a kubernetes service account for you, or use an existing kubernetes service account.
+
+### Creating a Workload Identity
+
+```hcl
+module "my-app-workload-identity" {
+  source    = "terraform-google-modules/terraform-google-kubernetes-engine/modules/workload-identity"
```
Comment (morgante) [inline modules/workload-identity/README.md]:
```suggestion
  source    = "terraform-google-modules/kubernetes-engine/google//modules/workload-identity"
```

File: examples/workload_identity/variables.tf (hunk 1 above)
Comment (bharathkkb) [reply examples/workload_identity/variables.tf]: Added default

File: examples/workload_identity/variables.tf (hunk 2 above)
Comment (bharathkkb) [reply examples/workload_identity/variables.tf]: Removed zones and defaulted to using regional cluster

File: modules/workload-identity/README.md (hunk 3 above)
Comment (bharathkkb) [reply modules/workload-identity/README.md]: fixed

**Pull request merged**